src/steps/node.rs

```diff
@@ -184,6 +184,79 @@ impl Yarn {
     }
 }
 
+struct Deno {
+    command: PathBuf,
+}
+
+impl Deno {
+    fn new(command: PathBuf) -> Self {
+        Self { command }
+    }
+
+    fn upgrade(&self, ctx: &ExecutionContext) -> Result<()> {
+        let mut args = vec![];
+
+        let version = ctx.config().deno_version();
+        if let Some(version) = version {
+            let bin_version = self.version()?;
+
+            if bin_version >= Version::new(2, 0, 0) {
+                args.push(version);
+            } else if bin_version >= Version::new(1, 6, 0) {
+                match version {
+                    "stable" => {}
+                    "rc" => {
+                        return Err(SkipStep("Deno v1.x cannot be upgraded to a release candidate".to_string()).into());
```

Comment:
Nit: this and line 225 are both called Deno v1.x, it might be better to make them different, e.g., `Deno (1.6.0-2.0.0)` and `Deno (1.0.0-1.6.0)`